C/common/reading_circularbuffer.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Fledge reading circular buffer class
+ *
+ * Copyright (c) 2025 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Mark Riddoch
+ */
+#include <reading_circularbuffer.h>
+
+using namespace std;
+
+/**
+ * Create a circular buffer of readings
+ *
+ * @param size	The numebr of items to retain oin the circular buffer
+ */
+ReadingCircularBuffer::ReadingCircularBuffer(unsigned int size) : m_size(size),
+       	m_insert(0), m_entries(0)
+{
+	m_readings.resize(size, NULL);
+}
+
+/**
+ * Destructor for the circular buffer
+ */
+ReadingCircularBuffer::~ReadingCircularBuffer()
+{
+	lock_guard<mutex> guard(m_mutex);
+	for (int i = 0; i < m_entries; i++)
+		m_readings[i] = NULL;
```

Comment:
If this is a destructor, don't you need to call _delete_ on every element of _m_readings_ rather than setting all reading points to NULL?

Reply:
They are shared pointers so setting this to NULL decreases the reference count and they will be deleted when the reference count becomes 0.